owncloudApp/src/main/java/com/owncloud/android/capabilities/datasource/OCRemoteCapabilitiesDataSource.kt

```diff
@@ -0,0 +1,35 @@
+/**
+ * ownCloud Android client application
+ *
+ * @author David González Verdugo
+ * Copyright (C) 2019 ownCloud GmbH.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.owncloud.android.capabilities.datasource
+
+import com.owncloud.android.lib.common.OwnCloudClient
+import com.owncloud.android.lib.common.operations.RemoteOperationResult
+import com.owncloud.android.lib.resources.status.GetRemoteCapabilitiesOperation
+import com.owncloud.android.lib.resources.status.RemoteCapability
+
+class OCRemoteCapabilitiesDataSource(
+    private val client: OwnCloudClient
+) : RemoteCapabilitiesDataSource {
+
+    override fun getCapabilities(getRemoteCapabilitiesOperation: GetRemoteCapabilitiesOperation)
+            : RemoteOperationResult<RemoteCapability> {
+        return getRemoteCapabilitiesOperation.execute(client)
```

Comment:
Could it be written this way?

`override fun getCapabilities(getRemoteCapabilitiesOperation: GetRemoteCapabilitiesOperation) = getRemoteCapabilitiesOperation.execute(client)`

Reply:
I tried but I got this warning

![Screenshot 2019-04-29 at 12 18 49](https://user-images.githubusercontent.com/15261363/56890170-004acd00-6a79-11e9-82d6-747fe8a8d72e.png)